src/app/basic_examples/test/page.mdx

```diff
@@ -0,0 +1,980 @@
+export const metadata = {
+  title: "Unit Testing • Stylus by Example",
+  description:
+    "How to write fast, deterministic unit tests for Stylus contracts with `stylus_sdk::testing::TestVM` and custom wrappers.",
+};
+
+{/* Begin Content */}
+# Unit Testing
+
+Stylus unit tests run **entirely in Rust**, with a fully-mocked host—no EVM, no RPC, no gas.
+You can exercise pure logic, mock all host contexts, inspect side-effects, and even extend the VM to suit your needs.
+
+
+## 1. HostAccess and `vm()`
+
+Every Stylus contract automatically implements the `HostAccess` trait, giving you a `.vm()` handle inside your methods.
+Use `self.vm()` instead of global helpers so your code works both in WASM and in native unit tests:
+
+```rust
+self.vm().msg_value()        // returns the mocked msg.value() in tests
+self.vm().msg_sender()       // returns the mocked msg.sender()
+self.vm().block_timestamp()  // returns the mocked block timestamp
+self.vm().call(&ctx, addr, &data) // invokes a mocked external call
+```
+
+In **production WASM** this maps to real host syscalls; in **native tests** it routes to `TestVM` or your custom host.
+
+## 2. Basic Unit Test
+
+With `stylus_sdk::testing::*` imported, write tests just like any Rust project. Below is a simple test suite for a counter contract that can be found at the bottom of the page.
+
+```rust
+#[cfg(test)]
+mod test {
+  use super::*;
+  use stylus_sdk::testing::*;
+
+  #[test]
+  fn test_counter_basic() {
+    // 1) Create a TestVM and contract
+    let vm = TestVM::default();
+    let mut c = Counter::from(&vm);
+
+    // 2) Assert initial state
+    assert_eq!(c.number(), U256::ZERO);
+
+    // 3) Call methods and assert logic
+    c.increment();
+    assert_eq!(c.number(), U256::ONE);
+
+    // 4) Mock msg.value() and test payable fn
+    vm.set_value(U256::from(5));
+    c.add_from_msg_value();
+    assert_eq!(c.number(), U256::from(6));
+  }
+}
+```
+
+**Explanation**
+
+* `TestVM::default()` seeds a clean in-memory VM. Use `TestVM::new()` to create a new VM instance for each test, ensuring isolation
```

Comment:
```suggestion
* `TestVM::default()` seeds a clean in-memory VM, use it to create a new VM instance for each test, ensuring isolation
```

I got a bit confused because it seemed that default and new did different things. However, I just checked the code, and default is just a wrapper on top of new. So, I recommend rewording it here.